api/src/main/java/org/openmrs/module/fhir2/FhirActivator.java

```diff
@@ -19,8 +25,58 @@
 @Slf4j
 public class FhirActivator extends BaseModuleActivator {
 	
+	/**
+	 * @return The immunizations encounter type
+	 */
+	public static EncounterType getImmunizationsEncounterTypeOrCreateIfMissing() {
```

Comment:
So here's the last thing I'm a little hesitant about with this: should we be automatically creating metadata like this? I think I'd be happier if we just bailed with an error in the case where the necessary metadata didn't exist. This is because while we're creating default mapping for FHIR resources, part of the goal is to have the implementations be reasonably easy to override locally to meet whatever requirements, so I don't love the idea of creating unnecessary encounters for people who decide to create their own implementation of the Immunization resource.

Reply:
I just changed that, the activator is now left alone ;-)